### PR TITLE
use native middleware buffer

### PR DESCRIPTION
**Pull request by zhakesi**

### CODE REVIEW

**Comment by fish9167** [comment]
@cocos-robot run test cases

**Review by SantyWang — APPROVED** [review]

File: native/cocos/2d/renderer/Batcher2d.cpp [hunk 1]
```diff
@@ -142,171 +143,191 @@ void Batcher2d::handlePostRender(RenderEntity* entity) {
         _stencilManager->enableMask();
     }
 }
+void CC_FORCE_INLINE Batcher2d::handleComponentDraw(RenderEntity* entity, RenderDrawInfo* drawInfo, Node* node) {
```
Comment (dumganhar) [inline native/cocos/2d/renderer/Batcher2d.cpp]: Put CC_FORCE_INLINE at the beginning.

Comment (zhakesi) [reply native/cocos/2d/renderer/Batcher2d.cpp]: done

File: native/cocos/2d/renderer/UIMeshBuffer.h [hunk 2]
```diff
@@ -72,6 +70,11 @@ class UIMeshBuffer final {
     void setDirty(bool dirty) const;
     inline uint32_t getFloatsPerVertex() const { return _meshBufferLayout->floatsPerVertex; }
     void setFloatsPerVertex(uint32_t floatsPerVertex);
+    inline bool getUseLinkData() const { return _useLinkData; }
+    inline void setUseLinkData(bool val) { _useLinkData = val;}
+    inline ccstd::vector<gfx::Attribute> attributes() const {
```
Comment (dumganhar) [inline native/cocos/2d/renderer/UIMeshBuffer.h]: Return `const &`

**Review by dumganhar — APPROVED** [review]

**Review by dumganhar — APPROVED** [review]

**Comment by pandamicro** [comment]
Web windows failed scenes were due to auto test framework issue, consider passed

**Pull request merged**